org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/CompletionTests14.java

```diff
@@ -518,4 +518,61 @@ public void testBug553097_5() throws JavaModelException {
 						requestor.getResults());
 
 	}
+
+	public void testGH667_CompletionOnRecordComponent_SimpleTypeName() throws JavaModelException {
+		this.workingCopies = new ICompilationUnit[2];
+		this.workingCopies[0] = getWorkingCopy(
+				"/Completion/src/Person.java",
+				"public record Person(Nam) {\n"
+						+ "}\n");
+		this.workingCopies[1] = getWorkingCopy(
+				"/Completion/src/Name.java",
+				"public class Name {\n"
+						+ "}\n");
+		CompletionTestsRequestor2 requestor = new CompletionTestsRequestor2(true, true, true, false);
+		String str = this.workingCopies[0].getSource();
+		String completeBehind = "Nam";
+		int cursorLocation = str.lastIndexOf(completeBehind) + completeBehind.length();
+		this.workingCopies[0].codeComplete(cursorLocation, requestor, this.wcOwner);
+		assertResults(
+				"Name[TYPE_REF]{Name, , LName;, null, null, null, null, [21, 24], 52}",
+				requestor.getResults());
+	}
+
+	public void testGH667_CompletionOnRecordComponent_QualifiedTypeName() throws JavaModelException {
+		this.workingCopies = new ICompilationUnit[1];
+		this.workingCopies[0] = getWorkingCopy(
+				"/Completion/src/Person.java",
+				"public record Person(pack2.P) {\n"
+						+ "}\n");
+		CompletionTestsRequestor2 requestor = new CompletionTestsRequestor2(true, true, true, false);
+		String str = this.workingCopies[0].getSource();
+		String completeBehind = "pack2.P";
+		int cursorLocation = str.lastIndexOf(completeBehind) + completeBehind.length();
+		this.workingCopies[0].codeComplete(cursorLocation, requestor, this.wcOwner);
+		assertResults(
+				"PX[TYPE_REF]{pack2.PX, pack2, Lpack2.PX;, null, null, null, null, [21, 28], 51}",
+				requestor.getResults());
+	}
+
+	public void testGH667_CompletionOnRecordComponent_VariableName() throws JavaModelException {
+		this.workingCopies = new ICompilationUnit[2];
+		this.workingCopies[0] = getWorkingCopy(
+				"/Completion/src/Person.java",
+				"public record Person(Name ) {\n"
+						+ "}\n");
+		this.workingCopies[1] = getWorkingCopy(
+				"/Completion/src/Name.java",
+				"public class Name {\n"
+						+ "}\n");
+		CompletionTestsRequestor2 requestor = new CompletionTestsRequestor2(true, true, true, false);
+		String str = this.workingCopies[0].getSource();
+		String completeBehind = "Name ";
+		int cursorLocation = str.lastIndexOf(completeBehind) + completeBehind.length();
+		this.workingCopies[0].codeComplete(cursorLocation, requestor, this.wcOwner);
+		assertResults(
+				"name[VARIABLE_DECLARATION]{name, null, LName;, null, null, name, null, [26, 26], 48}",
```

Comment:
I see that some other tests in this class look similar, but wouldn't it be interesting to actually document in the test, how the `relevance` adds up from individual constants? Cf., e.g., `testBug553097_?()`